Coolstore integration test failed with timeout issue

Step services "web-ui, inventory, catalog, cart, coolstore-gw" rollout successfully failed with timeout issue

[coolstore feature scenario](https://github.com/minishift/minishift/blob/master/test/integration/features/coolstore.feature#L36)
```
$ make integration GODOG_OPTS="-tags coolstore -format pretty"
.
.
Scenario: User creates a project and deploys CoolStore                                # features/coolstore.feature:19
    Given Minishift has state "Running"                                                 # integration_test.go:151 -> *Minishift.github.com/minishift/minishift/test/integration.shouldHaveState
    When executing "oc new-project coolstore" succeeds                                  # integration_test.go:648 -> github.com/minishift/minishift/test/integration.executingOcCommandSucceedsOrFails
    Then stdout should contain                                                          # integration_test.go:602 -> github.com/minishift/minishift/test/integration.commandReturnShouldContainContent
      """
      Now using project "coolstore"
      """
    When executing "oc create -f ./templates/coolstore.yaml" succeeds                   # integration_test.go:648 -> github.com/minishift/minishift/test/integration.executingOcCommandSucceedsOrFails
    Then stdout should contain                                                          # integration_test.go:602 -> github.com/minishift/minishift/test/integration.commandReturnShouldContainContent
      """
      template "coolstore" created
      """
    When executing "oc new-app --template=coolstore" succeeds                           # integration_test.go:648 -> github.com/minishift/minishift/test/integration.executingOcCommandSucceedsOrFails
    Then stdout should contain                                                          # integration_test.go:602 -> github.com/minishift/minishift/test/integration.commandReturnShouldContainContent
      """
      Success
      """
SIGQUIT: quit
PC=0x47f142 m=0 sigcode=0

goroutine 380 [syscall, 50 minutes]:
syscall.Syscall(0x0, 0x10, 0xc4201fe036, 0x5ca, 0x7f9f24d1e770, 0x223, 0xc4201fe000)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:18 +0x5 fp=0xc4201dd4b0 sp=0xc4201dd4a8
syscall.read(0x10, 0xc4201fe036, 0x5ca, 0x5ca, 0xf53380, 0x7f9f24d1e770, 0x0)
        /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:783 +0x55 fp=0xc4201dd508 sp=0xc4201dd4b0
syscall.Read(0x10, 0xc4201fe036, 0x5ca, 0x5ca, 0x42b6ff, 0xa7a4a8, 0xc4201dd5c0)
        /usr/lib/golang/src/syscall/syscall_unix.go:162 +0x49 fp=0xc4201dd550 sp=0xc4201dd508
os.(*File).read(0xc42068a0a8, 0xc4201fe036, 0x5ca, 0x5ca, 0x4dfeab, 0x98a900, 0x600)
        /usr/lib/golang/src/os/file_unix.go:165 +0x4f fp=0xc4201dd598 sp=0xc4201dd550
os.(*File).Read(0xc42068a0a8, 0xc4201fe036, 0x5ca, 0x5ca, 0x36, 0x0, 0x0)
        /usr/lib/golang/src/os/file.go:101 +0x76 fp=0xc4201dd600 sp=0xc4201dd598
bytes.(*Buffer).ReadFrom(0xc4200b43f0, 0xf214c0, 0xc42068a0a8, 0x7f9f24ca1050, 0xc4200b43f0, 0x2020202020202001)
        /usr/lib/golang/src/bytes/buffer.go:179 +0x160 fp=0xc4201dd670 sp=0xc4201dd600
io.copyBuffer(0xf1fd40, 0xc4200b43f0, 0xf214c0, 0xc42068a0a8, 0x0, 0x0, 0x0, 0x2020202020202020, 0x7365722020202020, 0x62203a656372756f)
        /usr/lib/golang/src/io/io.go:384 +0x2cb fp=0xc4201dd6d8 sp=0xc4201dd670
io.Copy(0xf1fd40, 0xc4200b43f0, 0xf214c0, 0xc42068a0a8, 0x7274203a73656d61, 0x202020200a0d6575, 0x2020202020202020)
        /usr/lib/golang/src/io/io.go:360 +0x68 fp=0xc4201dd738 sp=0xc4201dd6d8
os/exec.(*Cmd).writerDescriptor.func1(0x2020202020202020, 0x2020202020202020)
        /usr/lib/golang/src/os/exec/exec.go:254 +0x4d fp=0xc4201dd798 sp=0xc4201dd738
os/exec.(*Cmd).Start.func1(0xc4200ba9a0, 0xc42034ae40)
        /usr/lib/golang/src/os/exec/exec.go:371 +0x27 fp=0xc4201dd7d0 sp=0xc4201dd798
runtime.goexit()
        /usr/lib/golang/src/runtime/asm_amd64.s:2197 +0x1 fp=0xc4201dd7d8 sp=0xc4201dd7d0
created by os/exec.(*Cmd).Start
        /usr/lib/golang/src/os/exec/exec.go:372 +0x4e4

goroutine 1 [chan receive, 40 minutes]:
github.com/minishift/minishift/test/integration.(*Minishift).rolloutServicesSuccessfully(0xc4203b9d40, 0xc420271333, 0x2e, 0x10, 0xc4202f51a0)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/minishift.go:214 +0x264
github.com/minishift/minishift/test/integration.(*Minishift).(github.com/minishift/minishift/test/integration.rolloutServicesSuccessfully)-fm(0xc420271333, 0x2e, 0x0, 0x0)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/integration_test.go:177 +0x3e
reflect.Value.call(0x99cce0, 0xc4203e2c00, 0x13, 0xa50611, 0x4, 0xc42030d9e0, 0x1, 0x1, 0xc42021f1b0,0x98a640, ...)
        /usr/lib/golang/src/reflect/value.go:434 +0x91f
reflect.Value.Call(0x99cce0, 0xc4203e2c00, 0x13, 0xc42030d9e0, 0x1, 0x1, 0x0, 0x1, 0x13)
        /usr/lib/golang/src/reflect/value.go:302 +0xa4
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.(*StepDef).run(0xc4203c8f50, 0xa7a998, 0xc420507a78)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/stepdef.go:174 +0x275b
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.(*Suite).runStep(0xc4200250a0, 0xc4203900f0, 0x0, 0x0, 0x0, 0x0)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/suite.go:243 +0x15f
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.(*Suite).runSteps(0xc4200250a0, 0xc420379800, 0xa, 0x10, 0x0, 0x0, 0x0, 0x0)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/suite.go:250 +0x7d
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.(*Suite).runScenario(0xc4200250a0, 0xc420379700, 0x0, 0x0, 0x0)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/suite.go:367 +0x118
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.(*Suite).runFeature(0xc4200250a0, 0xc420383bc0)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/suite.go:342 +0x352
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.(*Suite).run(0xc4200250a0)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/suite.go:173 +0xa9
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.(*runner).run(0xc4203b3b80, 0x9)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/run.go:61 +0xa9
github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog.RunWithOptions(0xa56587, 0x9, 0xa7a8a0, 0x0, 0xa569d1, 0x9, 0xa52e30, 0x6, 0x0, 0xc4202daf30, ...)
        /home/amit/go/src/github.com/minishift/minishift/vendor/github.com/DATA-DOG/godog/run.go:120 +0x3d1
github.com/minishift/minishift/test/integration.TestMain(0xc420507f20)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/integration_test.go:103 +0x30f
main.main()
        github.com/minishift/minishift/test/integration/_test/_testmain.go:38 +0xf7

goroutine 17 [syscall, 60 minutes, locked to thread]:
runtime.goexit()
        /usr/lib/golang/src/runtime/asm_amd64.s:2197 +0x1

goroutine 26 [IO wait, 58 minutes]:
net.runtime_pollWait(0x7f9f24b8e700, 0x72, 0x4)
        /usr/lib/golang/src/runtime/netpoll.go:164 +0x59
net.(*pollDesc).wait(0xc420460618, 0x72, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_poll_runtime.go:75 +0x38
net.(*pollDesc).waitRead(0xc420460618, 0xc420106000, 0x800)
        /usr/lib/golang/src/net/fd_poll_runtime.go:80 +0x34
net.(*netFD).Read(0xc4204605b0, 0xc420106000, 0x800, 0x800, 0x0, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_unix.go:250 +0x1b7
net.(*conn).Read(0xc4202ba0d8, 0xc420106000, 0x800, 0x800, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/net/net.go:181 +0x70
crypto/tls.(*block).readFromUntil(0xc4202d4c30, 0xf20d00, 0xc4202ba0d8, 0x5, 0xc4202ba0d8, 0xc420044870)
        /usr/lib/golang/src/crypto/tls/conn.go:488 +0x98
crypto/tls.(*Conn).readRecord(0xc420267c00, 0xa7ba17, 0xc420267d20, 0x457220)
        /usr/lib/golang/src/crypto/tls/conn.go:590 +0xc4
crypto/tls.(*Conn).Read(0xc420267c00, 0xc420410000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/crypto/tls/conn.go:1134 +0x11d
net/http.(*persistConn).Read(0xc4202b4360, 0xc420410000, 0x1000, 0x1000, 0x5, 0x455320, 0xc4204be4e0)
        /usr/lib/golang/src/net/http/transport.go:1316 +0x14b
bufio.(*Reader).fill(0xc42007b2c0)
        /usr/lib/golang/src/bufio/bufio.go:97 +0x117
bufio.(*Reader).Peek(0xc42007b2c0, 0x1, 0xc420044be5, 0x1, 0x0, 0xc42007b320, 0x0)
        /usr/lib/golang/src/bufio/bufio.go:129 +0x67
net/http.(*persistConn).readLoop(0xc4202b4360)
        /usr/lib/golang/src/net/http/transport.go:1474 +0x196
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1117 +0xa35

goroutine 6 [syscall, 60 minutes]:
os/signal.signal_recv(0x0)
        /usr/lib/golang/src/runtime/sigqueue.go:116 +0x104
os/signal.loop()
        /usr/lib/golang/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.1
        /usr/lib/golang/src/os/signal/signal_unix.go:28 +0x41

goroutine 402 [syscall, 50 minutes]:
syscall.Syscall6(0xf7, 0x1, 0x6b23, 0xc4201dfb60, 0x1000004, 0x0, 0x0, 0x7f9f24d1e770, 0x0, 0xc4203b0220)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:44 +0x5
os.(*Process).blockUntilWaitable(0xc420382270, 0xc4201f2000, 0xc4203b0260, 0x6)
        /usr/lib/golang/src/os/wait_waitid.go:28 +0xa5
os.(*Process).wait(0xc420382270, 0xf52480, 0xc42012e200, 0xc420271260)
        /usr/lib/golang/src/os/exec_unix.go:22 +0x4d
os.(*Process).Wait(0xc420382270, 0x0, 0x0, 0xa7b5c0)
        /usr/lib/golang/src/os/exec.go:115 +0x2b
os/exec.(*Cmd).Wait(0xc4201f2000, 0x0, 0x0)
        /usr/lib/golang/src/os/exec/exec.go:435 +0x62
os/exec.(*Cmd).Run(0xc4201f2000, 0x7, 0xc4201e6090)
        /usr/lib/golang/src/os/exec/exec.go:280 +0x5c
github.com/minishift/minishift/test/integration/util.runCommand(0xc42032e000, 0x2f, 0xc4202711a0, 0x58, 0xe, 0xc4201dfec0, 0x7eb609, 0xc42012e0f0, 0xc4202e4001)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/util/util.go:55 +0x2ab
github.com/minishift/minishift/test/integration/util.(*OcRunner).RunCommand(0xc42012e0f0, 0xc42032e000, 0x2f, 0x1, 0x1, 0xc42032e000, 0x2f, 0x498843)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/util/util.go:166 +0x4f
github.com/minishift/minishift/test/integration.(*Minishift).checkServiceRolloutForSuccess(0xc4203b9d40, 0xc4202d4671, 0x7, 0xc420020700)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/minishift.go:181 +0x123
created by github.com/minishift/minishift/test/integration.(*Minishift).rolloutServicesSuccessfully
        /home/amit/go/src/github.com/minishift/minishift/test/integration/minishift.go:210 +0x177

goroutine 49 [syscall, 50 minutes]:
syscall.Syscall6(0xf7, 0x1, 0x6b13, 0xc4206a1b60, 0x1000004, 0x0, 0x0, 0x7f9f24d1d4b0, 0x0, 0xc42034ae80)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:44 +0x5
os.(*Process).blockUntilWaitable(0xc42032e8a0, 0xc4200ba9a0, 0xc42034aec0, 0x6)
        /usr/lib/golang/src/os/wait_waitid.go:28 +0xa5
os.(*Process).wait(0xc42032e8a0, 0xf52480, 0xc4202da640, 0xc420344720)
        /usr/lib/golang/src/os/exec_unix.go:22 +0x4d
os.(*Process).Wait(0xc42032e8a0, 0x0, 0x0, 0xa7b5c0)
        /usr/lib/golang/src/os/exec.go:115 +0x2b
os/exec.(*Cmd).Wait(0xc4200ba9a0, 0x0, 0x0)
        /usr/lib/golang/src/os/exec/exec.go:435 +0x62
os/exec.(*Cmd).Run(0xc4200ba9a0, 0x7, 0xc42032a5a0)
        /usr/lib/golang/src/os/exec/exec.go:280 +0x5c
github.com/minishift/minishift/test/integration/util.runCommand(0xc420291400, 0x31, 0xc4203446c0, 0x58, 0xe, 0xc4206a1ec0, 0x7eb609, 0xc4202da550, 0xc4200b8001)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/util/util.go:55 +0x2ab
github.com/minishift/minishift/test/integration/util.(*OcRunner).RunCommand(0xc4202da550, 0xc420291400, 0x31, 0x1, 0x1, 0xc420291400, 0x31, 0x498843)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/util/util.go:166 +0x4f
github.com/minishift/minishift/test/integration.(*Minishift).checkServiceRolloutForSuccess(0xc4203b9d40, 0xc4202d4667, 0x9, 0xc420020700)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/minishift.go:181 +0x123
created by github.com/minishift/minishift/test/integration.(*Minishift).rolloutServicesSuccessfully
        /home/amit/go/src/github.com/minishift/minishift/test/integration/minishift.go:210 +0x177

goroutine 27 [select, 58 minutes]:
net/http.(*persistConn).writeLoop(0xc4202b4360)
        /usr/lib/golang/src/net/http/transport.go:1704 +0x43a
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1118 +0xa5a

goroutine 38 [IO wait, 58 minutes]:
net.runtime_pollWait(0x7f9f24b8e640, 0x72, 0x6)
        /usr/lib/golang/src/runtime/netpoll.go:164 +0x59
net.(*pollDesc).wait(0xc420460068, 0x72, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_poll_runtime.go:75 +0x38
net.(*pollDesc).waitRead(0xc420460068, 0xc420017800, 0x800)
        /usr/lib/golang/src/net/fd_poll_runtime.go:80 +0x34
net.(*netFD).Read(0xc420460000, 0xc420017800, 0x800, 0x800, 0x0, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_unix.go:250 +0x1b7
net.(*conn).Read(0xc4202ba068, 0xc420017800, 0x800, 0x800, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/net/net.go:181 +0x70
crypto/tls.(*block).readFromUntil(0xc42001f140, 0xf20d00, 0xc4202ba068, 0x5, 0xc4202ba068, 0xc420045870)
        /usr/lib/golang/src/crypto/tls/conn.go:488 +0x98
crypto/tls.(*Conn).readRecord(0xc42007e000, 0xa7ba17, 0xc42007e120, 0x457220)
        /usr/lib/golang/src/crypto/tls/conn.go:590 +0xc4
crypto/tls.(*Conn).Read(0xc42007e000, 0xc420434000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/crypto/tls/conn.go:1134 +0x11d
net/http.(*persistConn).Read(0xc4202b4480, 0xc420434000, 0x1000, 0x1000, 0x5, 0x455320, 0xc420271ec0)
        /usr/lib/golang/src/net/http/transport.go:1316 +0x14b
bufio.(*Reader).fill(0xc420271d40)
        /usr/lib/golang/src/bufio/bufio.go:97 +0x117
bufio.(*Reader).Peek(0xc420271d40, 0x1, 0xc420045be5, 0x1, 0x0, 0xc420271da0, 0x0)
        /usr/lib/golang/src/bufio/bufio.go:129 +0x67
net/http.(*persistConn).readLoop(0xc4202b4480)
        /usr/lib/golang/src/net/http/transport.go:1474 +0x196
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1117 +0xa35

goroutine 39 [select, 58 minutes]:
net/http.(*persistConn).writeLoop(0xc4202b4480)
        /usr/lib/golang/src/net/http/transport.go:1704 +0x43a
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1118 +0xa5a

goroutine 114 [IO wait, 58 minutes]:
net.runtime_pollWait(0x7f9f24b8e580, 0x72, 0x7)
        /usr/lib/golang/src/runtime/netpoll.go:164 +0x59
net.(*pollDesc).wait(0xc420020538, 0x72, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_poll_runtime.go:75 +0x38
net.(*pollDesc).waitRead(0xc420020538, 0xc420425000, 0x800)
        /usr/lib/golang/src/net/fd_poll_runtime.go:80 +0x34
net.(*netFD).Read(0xc4200204d0, 0xc420425000, 0x800, 0x800, 0x0, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_unix.go:250 +0x1b7
net.(*conn).Read(0xc4204c2078, 0xc420425000, 0x800, 0x800, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/net/net.go:181 +0x70
crypto/tls.(*block).readFromUntil(0xc4203143f0, 0xf20d00, 0xc4204c2078, 0x5, 0xc4204c2078, 0x28)
        /usr/lib/golang/src/crypto/tls/conn.go:488 +0x98
crypto/tls.(*Conn).readRecord(0xc4205a4380, 0xa7ba17, 0xc4205a44a0, 0x457220)
        /usr/lib/golang/src/crypto/tls/conn.go:590 +0xc4
crypto/tls.(*Conn).Read(0xc4205a4380, 0xc420142000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/crypto/tls/conn.go:1134 +0x11d
net/http.(*persistConn).Read(0xc4200c9560, 0xc420142000, 0x1000, 0x1000, 0x5, 0x455320, 0xc42007b620)
        /usr/lib/golang/src/net/http/transport.go:1316 +0x14b
bufio.(*Reader).fill(0xc42007b4a0)
        /usr/lib/golang/src/bufio/bufio.go:97 +0x117
bufio.(*Reader).Peek(0xc42007b4a0, 0x1, 0xc4204aabe5, 0xc42012e201, 0x4, 0xc420322000, 0x0)
        /usr/lib/golang/src/bufio/bufio.go:129 +0x67
net/http.(*persistConn).readLoop(0xc4200c9560)
        /usr/lib/golang/src/net/http/transport.go:1474 +0x196
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1117 +0xa35

goroutine 115 [select, 58 minutes]:
net/http.(*persistConn).writeLoop(0xc4200c9560)
        /usr/lib/golang/src/net/http/transport.go:1704 +0x43a
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1118 +0xa5a

goroutine 403 [syscall, 50 minutes]:
syscall.Syscall6(0xf7, 0x1, 0x6b11, 0xc42069fb60, 0x1000004, 0x0, 0x0, 0x7f9f24d1d4b0, 0x0, 0xc42034a7a0)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:44 +0x5
os.(*Process).blockUntilWaitable(0xc42032e630, 0xc4200ba840, 0xc42034a7e0, 0x6)
        /usr/lib/golang/src/os/wait_waitid.go:28 +0xa5
os.(*Process).wait(0xc42032e630, 0xf52480, 0xc4202da2f0, 0xc420344240)
        /usr/lib/golang/src/os/exec_unix.go:22 +0x4d
os.(*Process).Wait(0xc42032e630, 0x0, 0x0, 0xa7b5c0)
        /usr/lib/golang/src/os/exec.go:115 +0x2b
os/exec.(*Cmd).Wait(0xc4200ba840, 0x0, 0x0)
        /usr/lib/golang/src/os/exec/exec.go:435 +0x62
os/exec.(*Cmd).Run(0xc4200ba840, 0x7, 0xc42032a240)
        /usr/lib/golang/src/os/exec/exec.go:280 +0x5c
github.com/minishift/minishift/test/integration/util.runCommand(0xc4202f4090, 0x2c, 0xc4203441e0, 0x58, 0xe, 0xc42069fec0, 0x7eb609, 0xc4202da230, 0xc42039a001)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/util/util.go:55 +0x2ab
github.com/minishift/minishift/test/integration/util.(*OcRunner).RunCommand(0xc4202da230, 0xc4202f4090, 0x2c, 0x1, 0x1, 0xc4202f4090, 0x2c, 0x498843)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/util/util.go:166 +0x4f
github.com/minishift/minishift/test/integration.(*Minishift).checkServiceRolloutForSuccess(0xc4203b9d40, 0xc4202d4679, 0x4, 0xc420020700)
        /home/amit/go/src/github.com/minishift/minishift/test/integration/minishift.go:181 +0x123
created by github.com/minishift/minishift/test/integration.(*Minishift).rolloutServicesSuccessfully
        /home/amit/go/src/github.com/minishift/minishift/test/integration/minishift.go:210 +0x177

goroutine 378 [syscall, 50 minutes]:
syscall.Syscall(0x0, 0x11, 0xc4201fe631, 0x5cf, 0x31, 0x200, 0x0)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x11, 0xc4201fe631, 0x5cf, 0x5cf, 0xc42020c000, 0x7f9f24d1e770, 0x0)
        /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x11, 0xc4201fe631, 0x5cf, 0x5cf, 0x42b6ff, 0xa7a4a8, 0xc4204a2dc0)
        /usr/lib/golang/src/syscall/syscall_unix.go:162 +0x49
os.(*File).read(0xc42068a060, 0xc4201fe631, 0x5cf, 0x5cf, 0x4dfeab, 0x98a900, 0x600)
        /usr/lib/golang/src/os/file_unix.go:165 +0x4f
os.(*File).Read(0xc42068a060, 0xc4201fe631, 0x5cf, 0x5cf, 0x31, 0x0, 0x0)
        /usr/lib/golang/src/os/file.go:101 +0x76
bytes.(*Buffer).ReadFrom(0xc4200b4230, 0xf214c0, 0xc42068a060, 0x7f9f24ca1050, 0xc4200b4230, 0xc4204a2f01)
        /usr/lib/golang/src/bytes/buffer.go:179 +0x160
io.copyBuffer(0xf1fd40, 0xc4200b4230, 0xf214c0, 0xc42068a060, 0x0, 0x0, 0x0, 0xc4200a84d8, 0x0, 0x0)
        /usr/lib/golang/src/io/io.go:384 +0x2cb
io.Copy(0xf1fd40, 0xc4200b4230, 0xf214c0, 0xc42068a060, 0x985000, 0xc4200a8480, 0xc4204a2fb0)
        /usr/lib/golang/src/io/io.go:360 +0x68
os/exec.(*Cmd).writerDescriptor.func1(0x985000, 0xc4200a8480)
        /usr/lib/golang/src/os/exec/exec.go:254 +0x4d
os/exec.(*Cmd).Start.func1(0xc4200ba840, 0xc42034a760)
        /usr/lib/golang/src/os/exec/exec.go:371 +0x27
created by os/exec.(*Cmd).Start
        /usr/lib/golang/src/os/exec/exec.go:372 +0x4e4

goroutine 247 [select, 58 minutes]:
net/http.(*persistConn).writeLoop(0xc4202b59e0)
        /usr/lib/golang/src/net/http/transport.go:1704 +0x43a
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1118 +0xa5a

goroutine 246 [IO wait, 58 minutes]:
net.runtime_pollWait(0x7f9f24b8e4c0, 0x72, 0x8)
        /usr/lib/golang/src/runtime/netpoll.go:164 +0x59
net.(*pollDesc).wait(0xc4200b5f08, 0x72, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_poll_runtime.go:75 +0x38
net.(*pollDesc).waitRead(0xc4200b5f08, 0xc4205ad000, 0x1000)
        /usr/lib/golang/src/net/fd_poll_runtime.go:80 +0x34
net.(*netFD).Read(0xc4200b5ea0, 0xc4205ad000, 0x1000, 0x1000, 0x0, 0xf22e80, 0xf1e3d0)
        /usr/lib/golang/src/net/fd_unix.go:250 +0x1b7
net.(*conn).Read(0xc4204c2430, 0xc4205ad000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/net/net.go:181 +0x70
net/http.(*persistConn).Read(0xc4202b59e0, 0xc4205ad000, 0x1000, 0x1000, 0x4, 0x455320, 0xc4203232c0)
        /usr/lib/golang/src/net/http/transport.go:1316 +0x14b
bufio.(*Reader).fill(0xc420345860)
        /usr/lib/golang/src/bufio/bufio.go:97 +0x117
bufio.(*Reader).Peek(0xc420345860, 0x1, 0xc4204a6be5, 0x1, 0x0, 0xc4203458c0, 0x0)
        /usr/lib/golang/src/bufio/bufio.go:129 +0x67
net/http.(*persistConn).readLoop(0xc4202b59e0)
        /usr/lib/golang/src/net/http/transport.go:1474 +0x196
created by net/http.(*Transport).dialConn
        /usr/lib/golang/src/net/http/transport.go:1117 +0xa35

goroutine 379 [syscall, 50 minutes]:
syscall.Syscall(0x0, 0x15, 0xc420126a00, 0x200, 0xc420126a00, 0x200, 0xc4202d4671)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x15, 0xc420126a00, 0x200, 0x200, 0xc42011e000, 0x7f9f24d1d960, 0x0)
        /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x15, 0xc420126a00, 0x200, 0x200, 0x42b6ff, 0xa7a4a8, 0xc42049cdc0)
        /usr/lib/golang/src/syscall/syscall_unix.go:162 +0x49
os.(*File).read(0xc42068a078, 0xc420126a00, 0x200, 0x200, 0x4dfeab, 0x98a900, 0x200)
        /usr/lib/golang/src/os/file_unix.go:165 +0x4f
os.(*File).Read(0xc42068a078, 0xc420126a00, 0x200, 0x200, 0x20, 0xc42033a0b0, 0xc42049ce60)
        /usr/lib/golang/src/os/file.go:101 +0x76
bytes.(*Buffer).ReadFrom(0xc4200b42a0, 0xf214c0, 0xc42068a078, 0x7f9f24ca1050, 0xc4200b42a0, 0x1)
        /usr/lib/golang/src/bytes/buffer.go:179 +0x160
io.copyBuffer(0xf1fd40, 0xc4200b42a0, 0xf214c0, 0xc42068a078, 0x0, 0x0, 0x0, 0xc4200a8358, 0x0, 0x0)
        /usr/lib/golang/src/io/io.go:384 +0x2cb
io.Copy(0xf1fd40, 0xc4200b42a0, 0xf214c0, 0xc42068a078, 0xc4202d4671, 0x7, 0x98a640)
        /usr/lib/golang/src/io/io.go:360 +0x68
os/exec.(*Cmd).writerDescriptor.func1(0x985000, 0xc4200a8300)
        /usr/lib/golang/src/os/exec/exec.go:254 +0x4d
os/exec.(*Cmd).Start.func1(0xc4200ba840, 0xc42034a7e0)
        /usr/lib/golang/src/os/exec/exec.go:371 +0x27
created by os/exec.(*Cmd).Start
        /usr/lib/golang/src/os/exec/exec.go:372 +0x4e4

goroutine 381 [syscall, 50 minutes]:
syscall.Syscall(0x0, 0x13, 0xc4200a4200, 0x200, 0xc4200a4200, 0x200, 0x755072657473616d)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x13, 0xc4200a4200, 0x200, 0x200, 0xc42011fc00, 0x7f9f24d1ec20, 0x0)
        /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x13, 0xc4200a4200, 0x200, 0x200, 0x42b6ff, 0xa7a4a8, 0xc4201dddc0)
        /usr/lib/golang/src/syscall/syscall_unix.go:162 +0x49
os.(*File).read(0xc42068a0c0, 0xc4200a4200, 0x200, 0x200, 0x4dfeab, 0x98a900, 0x200)
        /usr/lib/golang/src/os/file_unix.go:165 +0x4f
os.(*File).Read(0xc42068a0c0, 0xc4200a4200, 0x200, 0x200, 0x2020202020202020, 0x2020202020202020, 0xc4201dde60)
        /usr/lib/golang/src/os/file.go:101 +0x76
bytes.(*Buffer).ReadFrom(0xc4200b4460, 0xf214c0, 0xc42068a0c0, 0x7f9f24ca1050, 0xc4200b4460, 0x656b202020202001)
        /usr/lib/golang/src/bytes/buffer.go:179 +0x160
io.copyBuffer(0xf1fd40, 0xc4200b4460, 0xf214c0, 0xc42068a0c0, 0x0, 0x0, 0x0, 0x6966697472654364, 0x6e203a7365746163, 0x2020200a0d6c6c75)
        /usr/lib/golang/src/io/io.go:384 +0x2cb
io.Copy(0xf1fd40, 0xc4200b4460, 0xf214c0, 0xc42068a0c0, 0x202020200a0d3020, 0x2020202020202020, 0x2020202020202020)
        /usr/lib/golang/src/io/io.go:360 +0x68
os/exec.(*Cmd).writerDescriptor.func1(0x7561202020202020, 0x50656c6946746964)
        /usr/lib/golang/src/os/exec/exec.go:254 +0x4d
os/exec.(*Cmd).Start.func1(0xc4200ba9a0, 0xc42034aec0)
        /usr/lib/golang/src/os/exec/exec.go:371 +0x27
created by os/exec.(*Cmd).Start
        /usr/lib/golang/src/os/exec/exec.go:372 +0x4e4

goroutine 420 [syscall, 50 minutes]:
syscall.Syscall(0x0, 0x16, 0xc4206ab234, 0x5cc, 0x34, 0x200, 0x0)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x16, 0xc4206ab234, 0x5cc, 0x5cc, 0xc420140c00, 0x7f9f24d1e2c0, 0x0)
        /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x16, 0xc4206ab234, 0x5cc, 0x5cc, 0x42b6ff, 0xa7a4a8, 0xc4206b45c0)
        /usr/lib/golang/src/syscall/syscall_unix.go:162 +0x49
os.(*File).read(0xc4201ec020, 0xc4206ab234, 0x5cc, 0x5cc, 0x4dfeab, 0x98a900, 0x600)
        /usr/lib/golang/src/os/file_unix.go:165 +0x4f
os.(*File).Read(0xc4201ec020, 0xc4206ab234, 0x5cc, 0x5cc, 0x34, 0x0, 0x0)
        /usr/lib/golang/src/os/file.go:101 +0x76
bytes.(*Buffer).ReadFrom(0xc420460070, 0xf214c0, 0xc4201ec020, 0x7f9f24ca1050, 0xc420460070, 0x1)
        /usr/lib/golang/src/bytes/buffer.go:179 +0x160
io.copyBuffer(0xf1fd40, 0xc420460070, 0xf214c0, 0xc4201ec020, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/io/io.go:384 +0x2cb
io.Copy(0xf1fd40, 0xc420460070, 0xf214c0, 0xc4201ec020, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/io/io.go:360 +0x68
os/exec.(*Cmd).writerDescriptor.func1(0x0, 0x0)
        /usr/lib/golang/src/os/exec/exec.go:254 +0x4d
os/exec.(*Cmd).Start.func1(0xc4201f2000, 0xc4203b01e0)
        /usr/lib/golang/src/os/exec/exec.go:371 +0x27
created by os/exec.(*Cmd).Start
        /usr/lib/golang/src/os/exec/exec.go:372 +0x4e4

goroutine 421 [syscall, 50 minutes]:
syscall.Syscall(0x0, 0x18, 0xc42034dc00, 0x200, 0xc42034dc00, 0x200, 0x0)
        /usr/lib/golang/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x18, 0xc42034dc00, 0x200, 0x200, 0xc42006c000, 0x7f9f24d1de10, 0x0)
        /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x18, 0xc42034dc00, 0x200, 0x200, 0x42b6ff, 0xa7a4a8, 0xc4205aedc0)
        /usr/lib/golang/src/syscall/syscall_unix.go:162 +0x49
os.(*File).read(0xc4201ec038, 0xc42034dc00, 0x200, 0x200, 0x4dfeab, 0x98a900, 0x200)
        /usr/lib/golang/src/os/file_unix.go:165 +0x4f
os.(*File).Read(0xc4201ec038, 0xc42034dc00, 0x200, 0x200, 0x0, 0x0, 0xc4205aee60)
        /usr/lib/golang/src/os/file.go:101 +0x76
bytes.(*Buffer).ReadFrom(0xc4204600e0, 0xf214c0, 0xc4201ec038, 0x7f9f24ca1050, 0xc4204600e0, 0x1)
        /usr/lib/golang/src/bytes/buffer.go:179 +0x160
io.copyBuffer(0xf1fd40, 0xc4204600e0, 0xf214c0, 0xc4201ec038, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/io/io.go:384 +0x2cb
io.Copy(0xf1fd40, 0xc4204600e0, 0xf214c0, 0xc4201ec038, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/io/io.go:360 +0x68
os/exec.(*Cmd).writerDescriptor.func1(0x0, 0x0)
        /usr/lib/golang/src/os/exec/exec.go:254 +0x4d
os/exec.(*Cmd).Start.func1(0xc4201f2000, 0xc4203b0260)
        /usr/lib/golang/src/os/exec/exec.go:371 +0x27
created by os/exec.(*Cmd).Start
        /usr/lib/golang/src/os/exec/exec.go:372 +0x4e4

rax    0x0
rbx    0x0
rcx    0x47f144
rdx    0x5ca
rdi    0x10
rsi    0xc4201fe036
rbp    0xc4201dd4f8
rsp    0xc4201dd4a8
r8     0x0
r9     0x0
r10    0x0
r11    0x246
r12    0x0
r13    0xffffffee
r14    0x200
r15    0x10
rip    0x47f142
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
*** Test killed with quit: ran too long (1h1m0s).
FAIL    github.com/minishift/minishift/test/integration 3660.012s
make: *** [Makefile:176: integration] Error 1
$
```

This time coolstore deployment runs on stable branch 1.1.x and 1.0.x. On branch 1.1.x ends up with multiple image deployment failure like ```image pull time out error``` (same image can be pulled successfully inside the minishift vm). However branch 1.0.x is more stable than 1.1.x. All the pods are deployed successfully expect WEB-UI

```
npm ERR! Linux 4.4.41-boot2docker
npm ERR! argv "/opt/rh/rh-nodejs4/root/usr/bin/node" "/opt/rh/rh-nodejs4/root/usr/bin/npm" "install"
npm ERR! node v4.6.2
npm ERR! npm  v2.15.1
npm ERR! code ELIFECYCLE
npm ERR! coolstore-ui@1.0.0 postinstall: `bower install`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the coolstore-ui@1.0.0 postinstall script 'bower install'.
npm ERR! This is most likely a problem with the coolstore-ui package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     bower install
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs coolstore-ui
npm ERR! Or if that isn't available, you can get their info via:
npm ERR! 
npm ERR!     npm owner ls coolstore-ui
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /opt/app-root/src/npm-debug.log
error: build error: non-zero (13) exit code from centos/nodejs-4-centos7@sha256:8ef79a56d8fcf3558d89105df4315c0cb87ca6260325ff77e8a2a3000233cdb9 
```
It seems we are blocked on coolstore deployment cc_ @gbraad @praveenkumar @LalatenduMohanty @agajdosi . 

Added the findings in [coolstore repo]( jbossdemocentral/coolstore-microservice#74)

--- a/test/integration/features/stories/coolstore.feature
+++ b/test/integration/features/stories/coolstore.feature
@@ -1,73 +1,58 @@
-@coolstore @disabled
+@coolstore
 Feature: Cool Store
-  In order to test Minishift under load
-  I need to setup a test environment
-  Using CentOS and 4G of memory
-  After which I will import image streams
-  Finally I will deploy the Cool Store
-  
-  Scenario: User enables the 'xpaas' add-on and creates a Minishift instance
-    Given Minishift has state "Does Not Exist"
-      And image caching is disabled
-     When executing "minishift addons enable xpaas"
-     Then stdout should contain
-      """
-      Add-on 'xpaas' enabled
-      """
-     When executing "minishift start --memory 4096" succeeds
-     Then Minishift should have state "Running"
+  In order to test Minishift under load user starts it with 4G of memory,
+  imports xpaas image streams and finally deploys the Cool Store demo.
 
-  Scenario: User creates a project and deploys CoolStore
+  Scenario: User enables the 'xpaas' add-on
+    When executing "minishift addons enable xpaas"
+    Then stdout should contain "Add-on 'xpaas' enabled"
+
+  Scenario: User starts Minishift with 4GB of memory
+   Given Minishift has state "Does Not Exist"
+     And image caching is disabled
+    When executing "minishift start --memory 4096" succeeds
+    Then Minishift should have state "Running"
+
+  Scenario: User creates new project
    Given Minishift has state "Running"
     When executing "oc new-project coolstore" succeeds
     Then stdout should contain
      """
      Now using project "coolstore"
      """
-    When executing "oc create -f ./templates/coolstore.yaml" succeeds
+
+  Scenario: User adds Coolstore template
+    When executing "oc create -f ../../test/integration/templates/coolstore.yaml" succeeds
     Then stdout should contain
      """
-     template "coolstore" created
+     template.template.openshift.io "coolstore" created
      """
+
+  Scenario: User deploys new app from Coolstore template successfully
     When executing "oc new-app --template=coolstore" succeeds
-    Then stdout should contain
-     """
-     Success
-     """
-    When services "web-ui, inventory, catalog, cart, coolstore-gw" rollout successfully within "1200" seconds
-    Then executing "oc status -v" succeeds
-     And stdout should not contain
-     """
-     has failed
-     """
+     And services "web-ui, inventory, catalog, cart, coolstore-gw" rollout successfully within "1200" seconds
+    Then executing "oc status --suggest" succeeds
+     And stdout should not contain "has failed"
 
-  Scenario: User is able to interact with the deployed services from the API gateway
+  Scenario: Coolstore gateway exists
    Given Minishift has state "Running"
-    When setting scenario variable "coolstore-gw" to the stdout from executing "oc get pods -o name -l application=coolstore-gw"
+    When setting scenario variable "coolstore-gw" to the stdout from executing "oc get pods -o name -l app=coolstore-gw"
     Then scenario variable "coolstore-gw" should not be empty
-    When executing "oc rsh $(coolstore-gw) curl -sSL http://catalog:8080/api/products" succeeds
-    Then stdout should not contain
-     """
-     threw exception
-     """
-     And stdout should contain
-     """
-     Fedora
-     """
-    When executing "oc rsh $(coolstore-gw) curl -sSL http://inventory:8080/api/availability/329299" succeeds
-    Then stdout should contain
-     """
-     Raleigh
-     """
-    When executing "oc rsh $(coolstore-gw) curl -sSL http://cart:8080/api/cart/FOO" succeeds
-    Then stdout should contain
-     """
-     cartTotal
-     """
 
-  Scenario: User stops and deletes the Minishift instance
-    Given Minishift has state "Running"
-     When executing "minishift stop" succeeds
-     Then Minishift should have state "Stopped"
-     When executing "minishift delete --force" succeeds
-     Then Minishift should have state "Does Not Exist"
+  Scenario: User can list products over via Coolstore's API
+    When executing "oc rsh $(coolstore-gw) curl -sSL http://catalog:8080/api/products" succeeds
+    Then stdout should not contain "threw exception"
+     And stdout should contain "Fedora"
+
+   Scenario: User can check availability via Coolstore's API
+    When executing "oc rsh $(coolstore-gw) curl -sSL http://inventory:8080/api/availability/329299" succeeds
+    Then stdout should contain "Raleigh"
+
+   Scenario: User can check cart via Coolstore's API
+    When executing "oc rsh $(coolstore-gw) curl -sSL http://cart:8080/api/cart/FOO" succeeds
+    Then stdout should contain "cartTotal"
+
+  Scenario: User can delete the Minishift instance
+   Given Minishift has state "Running"
+    When executing "minishift delete --force" succeeds
+    Then Minishift should have state "Does Not Exist"

--- a/test/integration/templates/coolstore.yaml
+++ b/test/integration/templates/coolstore.yaml
@@ -9,6 +9,7 @@ metadata:
     iconClass: icon-java
     tags: microservice,jboss,spring
   name: coolstore
+message: "When using this template, the rating, review, turbine-server and hystrix-dashboard pods are slaced to zero by default to limit resource usage. Scale them up if you need these services or use the provisioning script instead."
 objects:
 - apiVersion: v1
   groupNames: null
@@ -26,7 +27,7 @@ objects:
   metadata:
     name: web-ui
     labels:
-      application: web-ui
+      app: web-ui
   spec:
     tags:
     - name: latest
@@ -35,7 +36,7 @@ objects:
   metadata:
     name: web-ui
     labels:
-      application: web-ui
+      app: web-ui
   spec:
     output:
       to:
@@ -49,10 +50,15 @@ objects:
       type: Git
     strategy:
       sourceStrategy:
+        env:
+        - name: NPM_MIRROR
+          value: ${NPM_MIRROR}
+        - name: BOWER_MIRROR
+          value: ${BOWER_MIRROR}
         from:
           kind: ImageStreamTag
           name: nodejs:4
-          namespace: openshift
+          namespace: ${IMAGE_STREAM_NAMESPACE}
       type: Source
     triggers:
     - imageChange: {}
@@ -63,11 +69,11 @@ objects:
   metadata:
     name: web-ui
     labels:
-      application: web-ui
+      app: web-ui
   spec:
     replicas: 1
     selector:
-      application: web-ui
+      app: web-ui
       deploymentconfig: web-ui
     strategy:
       resources: {}
@@ -75,7 +81,7 @@ objects:
     template:
       metadata:
         labels:
-          application: web-ui
+          app: web-ui
           deploymentconfig: web-ui
       spec:
         containers:
@@ -131,7 +137,6 @@ objects:
   metadata:
     labels:
       app: web-ui
-      application: web-ui
     name: web-ui
   spec:
     ports:
@@ -155,7 +160,7 @@ objects:
   metadata:
     name: coolstore-gw
     labels:
-      application: coolstore-gw
+      app: coolstore-gw
   spec: {}
   status:
     dockerImageRepository: ""
@@ -164,7 +169,7 @@ objects:
   metadata:
     name: coolstore-gw
     labels:
-      application: coolstore-gw
+      app: coolstore-gw
   spec:
     output:
       to:
@@ -187,7 +192,7 @@ objects:
         from:
           kind: ImageStreamTag
           name: fis-java-openshift:2.0
-          namespace: openshift
+          namespace: ${IMAGE_STREAM_NAMESPACE}
       type: Source
     triggers:
     - type: ConfigChange
@@ -198,7 +203,7 @@ objects:
   metadata:
     name: coolstore-gw
     labels:
-      application: coolstore-gw
+      app: coolstore-gw
   spec:
     replicas: 1
     selector:
@@ -209,7 +214,7 @@ objects:
     template:
       metadata:
         labels:
-          application: coolstore-gw
+          app: coolstore-gw
           deploymentconfig: coolstore-gw
         name: coolstore-gw
       spec:
@@ -254,7 +259,6 @@ objects:
   metadata:
     labels:
       app: coolstore-gw
-      application: coolstore-gw
       hystrix.enabled: "true"
     name: coolstore-gw
   spec:
@@ -269,7 +273,7 @@ objects:
   metadata:
     name: coolstore-gw
     labels:
-      application: coolstore-gw
+      app: coolstore-gw
   spec:
     to:
       kind: Service
@@ -280,7 +284,7 @@ objects:
   metadata:
     name: inventory
     labels:
-      application: inventory
+      app: inventory
   spec:
     tags:
     - name: latest
@@ -289,7 +293,7 @@ objects:
   metadata:
     name: inventory
     labels:
-      application: inventory
+      app: inventory
   spec:
     output:
       to:
@@ -308,8 +312,8 @@ objects:
           value: ${MAVEN_MIRROR_URL}
         from:
           kind: ImageStreamTag
-          name: jboss-eap70-openshift:1.4
-          namespace: openshift
+          name: jboss-eap70-openshift:1.5
+          namespace: ${IMAGE_STREAM_NAMESPACE}
       type: Source
     triggers:
     - type: ConfigChange
@@ -320,7 +324,7 @@ objects:
   metadata:
     name: inventory
     labels:
-      application: inventory
+      app: inventory
   spec:
     replicas: 1
     selector:
@@ -331,7 +335,7 @@ objects:
     template:
       metadata:
         labels:
-          application: inventory
+          app: inventory
           deploymentconfig: inventory
         name: inventory
       spec:
@@ -426,7 +430,6 @@ objects:
       service.alpha.openshift.io/dependencies: '[{"name":"inventory-postgresql","namespace":"","kind":"Service"}]'
     labels:
       app: inventory
-      application: inventory
     name: inventory
   spec:
     ports:
@@ -440,7 +443,8 @@ objects:
   metadata:
     name: inventory-postgresql
     labels:
-      application: inventory
+      app: inventory
+      comp-type: database
   spec:
     replicas: 1
     selector:
@@ -450,7 +454,8 @@ objects:
     template:
       metadata:
         labels:
-          application: inventory
+          app: inventory
+          comp-type: database
           deploymentconfig: inventory-postgresql
         name: inventory-postgresql
       spec:
@@ -497,7 +502,7 @@ objects:
         from:
           kind: ImageStreamTag
           name: postgresql:latest
-          namespace: openshift
+          namespace: ${IMAGE_STREAM_NAMESPACE}
       type: ImageChange
     - type: ConfigChange
 - apiVersion: v1
@@ -505,7 +510,6 @@ objects:
   metadata:
     labels:
       app: inventory
-      application: inventory
     name: inventory-postgresql
   spec:
     ports:
@@ -519,7 +523,7 @@ objects:
   metadata:
     name: catalog
     labels:
-      application: catalog
+      app: catalog
   spec:
     tags:
     - name: latest
@@ -545,8 +549,8 @@ objects:
           value: ${MAVEN_MIRROR_URL}
         from:
           kind: ImageStreamTag
-          name: jboss-webserver30-tomcat8-openshift:1.2
-          namespace: openshift
+          name: jboss-webserver31-tomcat8-openshift:1.0
+          namespace: ${IMAGE_STREAM_NAMESPACE}
       type: Source
     triggers:
     - type: ConfigChange
@@ -557,7 +561,7 @@ objects:
   metadata:
     name: catalog
     labels:
-      application: catalog
+      app: catalog
   spec:
     replicas: 1
     selector:
@@ -568,7 +572,7 @@ objects:
     template:
       metadata:
         labels:
-          application: catalog
+          app: catalog
           deploymentconfig: catalog
         name: catalog
       spec:
@@ -635,7 +639,6 @@ objects:
       service.alpha.openshift.io/dependencies: '[{"name":"catalog-mongodb","namespace":"","kind":"Service"}]'
     labels:
       app: catalog
-      application: catalog
     name: catalog
   spec:
     ports:
@@ -649,7 +652,6 @@ objects:
   metadata:
     labels:
       app: catalog
-      application: catalog
     name: catalog-mongodb
   spec:
     ports:
@@ -665,7 +667,8 @@ objects:
   kind: DeploymentConfig
   metadata:
     labels:
-      application: catalog
+      app: catalog
+      comp-type: database
     name: catalog-mongodb
   spec:
     replicas: 1
@@ -679,7 +682,8 @@ objects:
     template:
       metadata:
         labels:
-          application: catalog
+          app: catalog
+          comp-type: database
           deploymentconfig: catalog-mongodb
       spec:
         containers:
@@ -749,16 +753,289 @@ objects:
         from:
           kind: ImageStreamTag
           name: mongodb:3.2
-          namespace: openshift
+          namespace: ${IMAGE_STREAM_NAMESPACE}
       type: ImageChange
     - type: ConfigChange
+# Rating Service
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    name: rating
+    labels:
+      app: rating
+  spec:
+    tags:
+    - name: latest
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    name: rating
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: rating:latest
+    source:
+      contextDir: rating-service
+      git:
+        ref: ${GIT_REF}
+        uri: ${GIT_URI}
+      type: Git
+    strategy:
+      sourceStrategy:
+        env:
+        - name: MAVEN_MIRROR_URL
+          value: ${MAVEN_MIRROR_URL}
+        from:
+          kind: ImageStreamTag
+          name: redhat-openjdk18-openshift:1.1
+          namespace: ${IMAGE_STREAM_NAMESPACE}
+      type: Source
+    triggers:
+    - type: ConfigChange
+    - imageChange: {}
+      type: ImageChange
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: rating
+    labels:
+      app: rating
+  spec:
+    replicas: 0
+    selector:
+      deploymentconfig: rating
+    strategy:
+      resources: {}
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          app: rating
+          deploymentconfig: rating
+        name: rating
+      spec:
+        containers:
+        - env:
+          - name: KUBERNETES_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          image: rating
+          imagePullPolicy: Always
+          name: rating
+          ports:
+          - containerPort: 8778
+            name: jolokia
+            protocol: TCP
+          - containerPort: 8080
+            name: http
+            protocol: TCP
+          livenessProbe:
+            failureThreshold: 2
+            httpGet:
+              path: "/health/liveness"
+              port: 8080
+              scheme: HTTP
+            initialDelaySeconds: 60
+            periodSeconds: 10
+            successThreashold: 1
+            timeoutSeconds: 1
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: "/health/readiness"
+              port: 8080
+              scheme: HTTP
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            successThreashold: 1
+            timeoutSeconds: 1
+          resources:
+            limits:
+              memory: 1Gi
+            requests:
+              memory: 200Mi
+        terminationMessagePath: /dev/termination-log
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        securityContext: {}
+        terminationGracePeriodSeconds: 75
+    triggers:
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - rating
+        from:
+          kind: ImageStreamTag
+          name: rating:latest
+      type: ImageChange
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      service.alpha.openshift.io/dependencies: '[{"name":"rating-mongodb","namespace":"","kind":"Service"}]'
+    labels:
+      app: rating
+    name: rating
+  spec:
+    ports:
+    - port: 8080
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      deploymentconfig: rating
+- apiVersion: v1
+  kind: Service
+  metadata:
+    labels:
+      app: rating
+    name: rating-mongodb
+  spec:
+    ports:
+    - name: mongo
+      port: 27017
+      protocol: TCP
+      targetPort: 27017
+    selector:
+      deploymentconfig: rating-mongodb
+    sessionAffinity: None
+    type: ClusterIP
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    labels:
+      app: rating
+      comp-type: database
+    name: rating-mongodb
+  spec:
+    replicas: 0
+    selector:
+      deploymentconfig: rating-mongodb
+    strategy:
+      recreateParams:
+        post:
+          execNewPod:
+            command:
+              - /bin/sh
+              - -i
+              - -c
+              - env && while ! mongo ${RATING_MONGODB_SERVICE_HOST}:27017/$MONGODB_DATABASE -u $MONGODB_USER -p $MONGODB_PASSWORD --eval="$MONGODB_INIT" > /dev/null 2>&1; do echo "waiting for mongo ..."; sleep 5; done
+            containerName: rating-mongodb
+            env:
+            - name: MONGODB_INIT
+              value: db.ratings.insert({"_id":"329299","itemId":"329299","rating":5.0,"count":1});
+                db.ratings.insert({"_id":"329199","itemId":"329199","rating":1.0,"count":12});
+                db.ratings.insert({"_id":"165613","itemId":"165613","rating":2.3,"count":31});
+                db.ratings.insert({"_id":"165614","itemId":"165614","rating":3.0,"count":51});
+                db.ratings.insert({"_id":"165954","itemId":"165954","rating":4.0,"count":66});
+                db.ratings.insert({"_id":"444434","itemId":"444434","rating":5.0,"count":76});
+                db.ratings.insert({"_id":"444435","itemId":"444435","rating":4.0,"count":83});
+                db.ratings.insert({"_id":"444436","itemId":"444436","rating":3.0,"count":123});
+          failurePolicy: ignore
+        timeoutSeconds: 600
+      resources: {}
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          app: rating
+          comp-type: database
+          deploymentconfig: rating-mongodb
+      spec:
+        containers:
+        - env:
+          - name: KUBERNETES_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: MONGODB_USER
+            value: ${RATING_DB_USERNAME}
+          - name: MONGODB_PASSWORD
+            value: ${RATING_DB_PASSWORD}
+          - name: MONGODB_DATABASE
+            value: ratingdb
+          - name: MONGODB_ADMIN_PASSWORD
+            value: ${RATING_DB_PASSWORD}
+          image: mongodb
+          imagePullPolicy: IfNotPresent
+          livenessProbe:
+            failureThreshold: 10
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            successThreshold: 1
+            tcpSocket:
+              port: 27017
+            timeoutSeconds: 1
+          name: rating-mongodb
+          ports:
+          - containerPort: 27017
+            protocol: TCP
+          readinessProbe:
+            exec:
+              command:
+              - /bin/sh
+              - -i
+              - -c
+              - mongo 127.0.0.1:27017/$MONGODB_DATABASE -u $MONGODB_USER -p $MONGODB_PASSWORD
+                --eval="quit()"
+            failureThreshold: 10
+            initialDelaySeconds: 3
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          resources:
+            limits:
+              memory: 512Mi
+          securityContext:
+            capabilities: {}
+            privileged: false
+          terminationMessagePath: /dev/termination-log
+          volumeMounts:
+          - mountPath: /var/lib/mongodb/data
+            name: mongodb-data
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        securityContext: {}
+        terminationGracePeriodSeconds: 30
+        volumes:
+        - emptyDir: {}
+          name: mongodb-data
+    test: false
+    triggers:
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - rating-mongodb
+        from:
+          kind: ImageStreamTag
+          name: mongodb:3.2
+          namespace: ${IMAGE_STREAM_NAMESPACE}
+      type: ImageChange
+    - type: ConfigChange
+
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: rating-config
+    labels:
+      app: rating
+  data:
+    rating-config.yaml: |-
+      rating.http.port: 8080
+      connection_string: mongodb://rating-mongodb:27017
+      db_name: ratingdb
+      username: ${RATING_DB_USERNAME}
+      password: ${RATING_DB_PASSWORD}
+
 # Cart Service
 - apiVersion: v1
   kind: ImageStream
   metadata:
     name: cart
     labels:
-      application: cart
+      app: cart
   spec:
     tags:
     - name: latest
@@ -767,7 +1044,7 @@ objects:
   metadata:
     name: cart
     labels:
-      application: cart
+      app: cart
   spec:
     output:
       to:
@@ -786,8 +1063,8 @@ objects:
           value: ${MAVEN_MIRROR_URL}
         from:
           kind: ImageStreamTag
-          name: redhat-openjdk18-openshift:1.0
-          namespace: openshift
+          name: redhat-openjdk18-openshift:1.1
+          namespace: ${IMAGE_STREAM_NAMESPACE}
       type: Source
     triggers:
     - type: ConfigChange
@@ -798,7 +1075,7 @@ objects:
   metadata:
     name: cart
     labels:
-      application: cart
+      app: cart
   spec:
     replicas: 1
     selector:
@@ -809,7 +1086,7 @@ objects:
     template:
       metadata:
         labels:
-          application: cart
+          app: cart
           deploymentconfig: cart
         name: cart
       spec:
@@ -817,6 +1094,14 @@ objects:
         - env:
           - name: CATALOG_ENDPOINT
             value: "http://catalog:8080"
+          - name: PRICING_ENDPOINT
+            value: "http://pricing:8080"
+          - name: KIE_CONTAINER_DEPLOYMENT
+            value: ${KIE_CONTAINER_DEPLOYMENT}
+          - name: KIE_SERVER_USER
+            value: ${KIE_SERVER_USER}
+          - name: KIE_SERVER_PASSWORD
+            value: ${KIE_SERVER_PASSWORD}
           image: cart
           imagePullPolicy: Always
           livenessProbe:
@@ -825,7 +1110,7 @@ objects:
               path: /health
               port: 8080
               scheme: HTTP
-            initialDelaySeconds: 15
+            initialDelaySeconds: 30
             periodSeconds: 5
             successThreshold: 1
             timeoutSeconds: 1
@@ -846,7 +1131,7 @@ objects:
               path: /health
               port: 8080
               scheme: HTTP
-            initialDelaySeconds: 15
+            initialDelaySeconds: 30
             periodSeconds: 5
             successThreshold: 1
             timeoutSeconds: 1
@@ -875,7 +1160,6 @@ objects:
   metadata:
     labels:
       app: cart
-      application: cart
     name: cart
   spec:
     ports:
@@ -884,12 +1168,406 @@ objects:
       targetPort: 8080
     selector:
       deploymentconfig: cart
+# Pricing Rules Service
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    name: pricing
+  spec:
+    tags:
+    - name: latest
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    name: pricing
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: pricing:latest
+    source:
+      contextDir: pricing-service
+      git:
+        ref: ${GIT_REF}
+        uri: ${GIT_URI}
+      type: Git
+    strategy:
+      sourceStrategy:
+        env:
+        - name: KIE_CONTAINER_DEPLOYMENT
+          value: ${KIE_CONTAINER_DEPLOYMENT}
+        - name: MAVEN_MIRROR_URL
+          value: ${MAVEN_MIRROR_URL}
+        from:
+          kind: ImageStreamTag
+          name: jboss-decisionserver64-openshift:1.0
+          namespace: ${IMAGE_STREAM_NAMESPACE}
+      type: Source
+    triggers:
+    - type: ConfigChange
+    - imageChange: {}
+      type: ImageChange
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: pricing
+  spec:
+    replicas: 0
+    selector:
+      deploymentconfig: pricing
+    strategy:
+      resources: {}
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          application: pricing
+          deploymentconfig: pricing
+        name: pricing
+      spec:
+        containers:
+        - env:
+          - name: MQ_CLUSTER_PASSWORD
+            value: 7mzX0pLV03
+          - name: JGROUPS_CLUSTER_PASSWORD
+            value: CqUo3fYDTv
+          - name: AUTO_DEPLOY_EXPLODED
+            value: "false"
+          - name: KIE_CONTAINER_DEPLOYMENT
+            value: ${KIE_CONTAINER_DEPLOYMENT}
+          - name: KIE_SERVER_USER
+            value: ${KIE_SERVER_USER}
+          - name: KIE_SERVER_PASSWORD
+            value: ${KIE_SERVER_PASSWORD}
+          - name: OPENSHIFT_KUBE_PING_LABELS
+            value: application=pricing-service
+          - name: OPENSHIFT_KUBE_PING_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          image: pricing
+          imagePullPolicy: Always
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                - /opt/eap/bin/jboss-cli.sh
+                - -c
+                - :shutdown(timeout=60)
+          livenessProbe:
+            failureThreshold: 5
+            httpGet:
+              path: /kie-server/docs/
+              port: 8080
+              scheme: HTTP
+            initialDelaySeconds: 120
+            periodSeconds: 5
+            successThreshold: 1
+            timeoutSeconds: 5
+          name: pricing
+          ports:
+          - containerPort: 8778
+            name: jolokia
+            protocol: TCP
+          - containerPort: 8080
+            name: http
+            protocol: TCP
+          - containerPort: 8888
+            name: ping
+            protocol: TCP
+          readinessProbe:
+            failureThreshold: 10
+            httpGet:
+              path: /kie-server/docs/
+              port: 8080
+              scheme: HTTP
+            initialDelaySeconds: 15
+            periodSeconds: 5
+            successThreshold: 1
+            timeoutSeconds: 1
+          resources:
+            limits:
+              memory: 600Mi
+            requests:
+              memory: 200Mi
+          terminationMessagePath: /dev/termination-log
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        securityContext: {}
+        terminationGracePeriodSeconds: 75
+    triggers:
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - pricing
+        from:
+          kind: ImageStreamTag
+          name: pricing:latest
+      type: ImageChange
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      description: The pricing service's http port.
+    labels:
+      app: pricing
+    name: pricing
+  spec:
+    ports:
+    - port: 8080
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      deploymentconfig: pricing
+
+# Review Service
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    name: review
+    labels:
+      app: review
+  spec:
+    tags:
+    - name: latest
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    name: review
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: review:latest
+    source:
+      contextDir: review-service
+      git:
+        ref: ${GIT_REF}
+        uri: ${GIT_URI}
+      type: Git
+    strategy:
+      sourceStrategy:
+        env:
+        - name: MAVEN_MIRROR_URL
+          value: ${MAVEN_MIRROR_URL}
+        from:
+          kind: ImageStreamTag
+          name: redhat-openjdk18-openshift:1.1
+          namespace: ${IMAGE_STREAM_NAMESPACE}
+      type: Source
+    triggers:
+    - type: ConfigChange
+    - imageChange: {}
+      type: ImageChange
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: review
+    labels:
+      app: review
+  spec:
+    replicas: 0
+    selector:
+      deploymentconfig: review
+    strategy:
+      resources: {}
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          app: review
+          deploymentconfig: review
+        name: review
+      spec:
+        containers:
+        - env:
+          - name: KUBERNETES_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: JAVA_OPTIONS
+            value: "-Dswarm.project.stage=production -Dswarm.project.stage.file=file:///app/config/review-config.yml"
+          volumeMounts:
+            - name: config
+              mountPath: /app/config
+          image: review
+          imagePullPolicy: Always
+          name: review
+          ports:
+          - containerPort: 8778
+            name: jolokia
+            protocol: TCP
+          - containerPort: 8080
+            name: http
+            protocol: TCP
+          livenessProbe:
+            failureThreshold: 2
+            httpGet:
+              path: "/health"
+              port: 8080
+              scheme: HTTP
+            initialDelaySeconds: 60
+            periodSeconds: 10
+            successThreashold: 1
+            timeoutSeconds: 1
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: "/health"
+              port: 8080
+              scheme: HTTP
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            successThreashold: 1
+            timeoutSeconds: 1
+          resources:
+            limits:
+              memory: 1Gi
+            requests:
+              memory: 200Mi
+        volumes:
+          - configMap:
+              name: review-config
+              items:
+              - key: "review-config.yml"
+                path: "review-config.yml"
+            name: config
+        terminationMessagePath: /dev/termination-log
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        securityContext: {}
+        terminationGracePeriodSeconds: 75
+    triggers:
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - review
+        from:
+          kind: ImageStreamTag
+          name: review:latest
+      type: ImageChange
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      service.alpha.openshift.io/dependencies: '[{"name":"review-postgresql","namespace":"","kind":"Service"}]'
+    labels:
+      app: review
+    name: review
+  spec:
+    ports:
+    - port: 8080
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      deploymentconfig: review
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: review-postgresql
+    labels:
+      comp-type: database
+      app: review
+  spec:
+    replicas: 0
+    selector:
+      deploymentconfig: review-postgresql
+    strategy:
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          app: review
+          comp-type: database
+          deploymentconfig: review-postgresql
+        name: review-postgresql
+      spec:
+        containers:
+        - env:
+          - name: POSTGRESQL_USER
+            value: ${REVIEW_DB_USERNAME}
+          - name: POSTGRESQL_PASSWORD
+            value: ${REVIEW_DB_PASSWORD}
+          - name: POSTGRESQL_DATABASE
+            value: review
+          image: postgresql
+          imagePullPolicy: Always
+          name: review-postgresql
+          volumeMounts:
+          - mountPath: /var/lib/pgsql/data
+            name: review-postgresql-data
+          ports:
+          - containerPort: 5432
+            protocol: TCP
+          livenessProbe:
+            initialDelaySeconds: 30
+            tcpSocket:
+              port: 5432
+            timeoutSeconds: 1
+          readinessProbe:
+            exec:
+              command:
+              - /bin/sh
+              - -i
+              - -c
+              - psql -h 127.0.0.1 -U $POSTGRESQL_USER -q -d $POSTGRESQL_DATABASE -c 'SELECT 1'
+            initialDelaySeconds: 5
+            timeoutSeconds: 1
+        terminationGracePeriodSeconds: 60
+        volumes:
+        - name: review-postgresql-data
+          emptyDir: {}
+    triggers:
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - review-postgresql
+        from:
+          kind: ImageStreamTag
+          name: postgresql:latest
+          namespace: ${IMAGE_STREAM_NAMESPACE}
+      type: ImageChange
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    labels:
+      app: review
+    name: review-postgresql
+  spec:
+    ports:
+    - port: 5432
+      targetPort: 5432
+    selector:
+      deploymentconfig: review-postgresql
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: review-config
+    labels:
+      app: review
+  data:
+    review-config.yml: |-
+      project:
+        stage: production
+      swarm:
+        datasources:
+          data-sources:
+            ReviewDS:
+              driver-name: postgresql
+              connection-url: jdbc:postgresql://review-postgresql:5432/review
+              user-name: ${REVIEW_DB_USERNAME}
+              password: ${REVIEW_DB_PASSWORD}
+
 # Netflix Turbine Server
 - apiVersion: v1
   kind: ImageStream
   metadata:
     labels:
-      application: turbine-server
+      component: turbine-server
+      app: netflix
     name: turbine-server
   spec:
     dockerImageRepository: docker.io/siamaksade/turbine-server # temporary till fabric8 images remove privillegd requirement
@@ -902,20 +1580,23 @@ objects:
   kind: DeploymentConfig
   metadata:
     labels:
-      application: turbine-server
+      component: turbine-server
+      app: netflix
     name: turbine-server
   spec:
-    replicas: 1
+    replicas: 0
     selector:
-      application: turbine-server
       deploymentconfig: turbine-server
+      component: turbine-server
+      app: netflix
     strategy:
       type: Recreate
     template:
       metadata:
         labels:
-          application: turbine-server
           deploymentconfig: turbine-server
+          component: turbine-server
+          app: netflix
       spec:
         containers:
         - image: 'turbine-server'
@@ -959,32 +1640,29 @@ objects:
           kind: ImageStreamTag
           name: turbine-server:1.0.27
       type: ImageChange
-  status:
-    observedGeneration: 2
-    replicas: 1
-    unavailableReplicas: 1
-    updatedReplicas: 1
 - apiVersion: v1
   kind: Service
   metadata:
     labels:
-      app: turbine-server
-      application: turbine-server
+      component: turbine-server
+      app: netflix
     name: turbine-server
   spec:
     ports:
-    - name: 8080-tcp
-      port: 8080
+    - name: 80-tcp
+      port: 80
       protocol: TCP
       targetPort: 8080
     selector:
-      application: turbine-server
       deploymentconfig: turbine-server
+      component: turbine-server
+      app: netflix
 - apiVersion: v1
   kind: Route
   metadata:
     labels:
-      application: turbine-server
+      component: turbine-server
+      app: netflix
     name: turbine-server
   spec:
     port:
@@ -998,7 +1676,8 @@ objects:
   kind: ImageStream
   metadata:
     labels:
-      application: hystrix-dashboard
+      component: hystrix-dashboard
+      app: netflix
     name: hystrix-dashboard
   spec:
     dockerImageRepository: docker.io/siamaksade/hystrix-dashboard # temporary till fabric8 images remove privillegd requirement
@@ -1011,20 +1690,23 @@ objects:
   kind: DeploymentConfig
   metadata:
     labels:
-      application: hystrix-dashboard
+      component: hystrix-dashboard
+      app: netflix
     name: hystrix-dashboard
   spec:
-    replicas: 1
+    replicas: 0
     selector:
-      application: hystrix-dashboard
       deploymentconfig: hystrix-dashboard
+      component: hystrix-dashboard
+      app: netflix
     strategy:
       type: Recreate
     template:
       metadata:
         labels:
-          application: hystrix-dashboard
           deploymentconfig: hystrix-dashboard
+          component: hystrix-dashboard
+          app: netflix
       spec:
         containers:
         - image: 'hystrix-dashboard'
@@ -1072,8 +1754,8 @@ objects:
   kind: Service
   metadata:
     labels:
-      app: hystrix-dashboard
-      application: hystrix-dashboard
+      component: hystrix-dashboard
+      app: netflix
     annotations:
       service.alpha.openshift.io/dependencies: '[{"name":"turbine-server","namespace":"","kind":"Service"}]'
     name: hystrix-dashboard
@@ -1084,13 +1766,15 @@ objects:
       protocol: TCP
       targetPort: 8080
     selector:
-      application: hystrix-dashboard
       deploymentconfig: hystrix-dashboard
+      component: hystrix-dashboard
+      app: netflix
 - apiVersion: v1
   kind: Route
   metadata:
     labels:
-      application: hystrix-dashboard
+      component: hystrix-dashboard
+      app: netflix
     name: hystrix-dashboard
   spec:
     port:
@@ -1108,10 +1792,16 @@ parameters:
 - displayName: Git branch/tag reference
   name: GIT_REF
   required: true
-  value: 1.0.x
+  value: stable-ocp-3.10
 - description: Maven mirror url. If nexus is deployed locally, use nexus url (e.g. http://nexus.ci:8081/content/groups/public/)
   displayName: Maven mirror url
   name: MAVEN_MIRROR_URL
+- description: NPM mirror url. If nexus is deployed locally, use nexus url (e.g. http://nexus.ci:8081/repository/npm/)
+  displayName: NPM mirror url
+  name: NPM_MIRROR
+- description: Bower mirror url. If nexus is deployed locally, use nexus url (e.g. http://nexus.ci:8081/repository/bower-mirror/)
+  displayName: Bower mirror url
+  name: BOWER_MIRROR
 - description: Inventory Service database user name
   displayName: Inventory Database username
   from: user[a-zA-Z0-9]{3}
@@ -1135,4 +1825,46 @@ parameters:
   from: '[a-zA-Z0-9]{8}'
   generate: expression
   name: CATALOG_DB_PASSWORD
+  required: true
+- description: Decision Server Username
+  name: KIE_SERVER_USER
+  required: true
+  value: brmsAdmin
+- description: Decision Server Password
+  from: '[a-zA-Z]{6}[0-9]{1}!'
+  generate: expression
+  name: KIE_SERVER_PASSWORD
+  required: true
+- description: Decision Server Coolstore Rules Container Spec
+  name: KIE_CONTAINER_DEPLOYMENT
+  required: true
+  value: CoolStoreRulesContainer=com.redhat:coolstore:2.0.0
+- description: Rating Service database user name
+  displayName: Rating Database username
+  from: user[a-zA-Z0-9]{3}
+  generate: expression
+  name: RATING_DB_USERNAME
+  required: true
+- description: Rating Service database user password
+  displayName: Rating Database password
+  from: '[a-zA-Z0-9]{8}'
+  generate: expression
+  name: RATING_DB_PASSWORD
+  required: true
+- description: Review Service database user name
+  displayName: Review Database username
+  from: user[a-zA-Z0-9]{3}
+  generate: expression
+  name: REVIEW_DB_USERNAME
+  required: true
+- description: Review Service database user password
+  displayName: Review Database password
+  from: '[a-zA-Z0-9]{8}'
+  generate: expression
+  name: REVIEW_DB_PASSWORD
+  required: true
+- description: Namespace in which the ImageStreams for Red Hat Middleware images are installed. These ImageStreams are normally installed in the openshift namespace. You should only need to modify this if you've installed the ImageStreams in a different namespace/project.
+  displayName: ImageStream Namespace
+  value: openshift
+  name: IMAGE_STREAM_NAMESPACE
   required: true

